### PR TITLE
Fix(rollups): Fix splits in roll-up

### PR DESCRIPTION
**Pull request by ahsanbarkati**
There was an issue with split creation in roll-up. When a part `p` is split 
into two parts say `p1` and `p2` then the first UID of `p2` should be greater
than the last UID of p1. This was being violated because of removing UIDs 
until `maxInt64` instead of `maxUint64` from `p` in order to create `p1`. 

This PR fixes this issue and also removes some unwanted codes. 

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/dgraph/7609)

### CODE REVIEW

File: posting/list.go [hunk 1]
```diff
@@ -1191,6 +1193,8 @@ func (l *List) encode(out *rollupOutput, readTs uint64, split bool) error {
 		out.parts[startUid] = plist
 	}
 
+	out.plist.Bitmap = codec.ToBytes(bm)
```
Comment (ahsanbarkati) [inline posting/list.go:1196]: Need to figure out why things were working without this change.

Comment (ahsanbarkati) [reply posting/list.go:1196]: Add a comment here, we don't set plist here. It is set by the finalize() function at the end.

File: posting/list.go [hunk 2]
```diff
@@ -963,12 +962,13 @@ func (l *List) ToBackupPostingList(bl *pb.BackupPostingList, alloc *z.Allocator)
 	// out is only nil when the list's minTs is greater than readTs but readTs
 	// is math.MaxUint64 so that's not possible. Assert that's true.
 	x.AssertTrue(out != nil)
-	// defer out.free()
 
 	ol := out.plist
 	bm := roaring64.New()
-	if err := bm.UnmarshalBinary(ol.Bitmap); err != nil {
-		return nil, errors.Wrapf(err, "failed when unmarshal binary bitmap")
+	if ol.Bitmap != nil {
```
Comment (jarifibrahim) [inline posting/list.go:968]: Is this nil check necessary?

**Review by jarifibrahim — APPROVED** [review]
The PR looks good to me. Can we please add a test for this @ahsanbarkati ?

File: posting/list.go (hunk 2 above)
Comment (ahsanbarkati) [reply posting/list.go:968]: Yes. In case of empty list `ol.Bitmap` turns out to be `nil` and `bm.UnmarshalBinray()` panics.

**Pull request merged**